cmd/mv-main.go

```diff
@@ -210,9 +211,14 @@ func mainMove(cliCtx *cli.Context) error {
 		args := cliCtx.Args()
 		srcURL := args.Get(0)
 		dstURL := args.Get(1)
-		if srcURL == dstURL {
+		// case: mc mv play/mybucket/path play/mybucket
+		if strings.HasPrefix(srcURL, dstURL) {
 			fatalIf(errDummy().Trace(), fmt.Sprintf("Source and destination urls cannot be the same: %v.", srcURL))
 		}
+		// case: mc mv play/mybucket/path play/mybucket/path/path2
+		if strings.HasPrefix(dstURL, srcURL) {
+			fatalIf(errDummy().Trace(), fmt.Sprintf("Source %s contains destination: %v.", srcURL, dstURL))
+		}
```

Comment:
These two conditions can be merged in one with `||` may be and we can have generic message like `Source and destination cannot be same`